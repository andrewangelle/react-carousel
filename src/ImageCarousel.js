import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group'
import './index.css';
import Preload from 'react-preload';

const loadingIndicator = (<div>Loading...</div>);
    
const images = [
      'https://unsplash.it/500/?random',
      'https://unsplash.it/501/?random',
      'https://unsplash.it/502/?random',
      'https://unsplash.it/503/?random',
      'https://unsplash.it/504/?random',
      'https://unsplash.it/505/?random'
      ];


export default class ImageCarousel extends React.Component {
  constructor(props) {
    super(props);

    this.state={
       currentIndex: 0

     }
  };

   goToPrevImage() {
    const { currentIndex }=this.state;
    const nextIndex = currentIndex === 0 ? 
          this.props.images.length - 1 : currentIndex - 1;
   
     this.setState({ currentIndex : nextIndex })

   }

   goToNextImage() {
    const { currentIndex }=this.state;
    const nextIndex = currentIndex === this.props.images.length - 1 ? 
          0 : currentIndex + 1;

     this.setState({ currentIndex : nextIndex })
   }

  render(){

    var style={
      position: "relative",
      width: this.props.width
    }

    return (
      <div className="container" style={style}>

        <Preload
            loadingIndicator={loadingIndicator}
            images={images}
            autoResolveDelay={3000}
            onError={this._handleImageLoadError}
            onSuccess={this._handleImageLoadSuccess}
            resolveOnError={true}
            mountChildren={true}>

        <ReactCSSTransitionGroup
          transitionName='current'
          transitionAppear={true}
          transitionAppearTimeout={500}
          transitionEnterTimeout={500}
          transitionLeaveTimeout={300}>

        <img 
          className='carousel-image' 
          key={this.state.currentIndex} 
          src={this.props.images[this.state.currentIndex]} />

          <button onClick={this.goToPrevImage.bind(this)}> Next </button>
          <button className="next" onClick={this.goToNextImage.bind(this)}> Prev </button>


        </ReactCSSTransitionGroup>
        </Preload>


      </div>

     );
  }
}
