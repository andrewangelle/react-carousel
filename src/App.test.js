import React from 'react';
import ImageCarousel from './ImageCarousel.js';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';





describe('ImageCarousel', () => {
	it('renders without crashing', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];


		const tree = renderer.create(
			<ImageCarousel images={images} />
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

		it('has images array that contains images', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];

		const componenet = shallow(
			<ImageCarousel images={images} />
		);

		expect(images.length).toBeGreaterThan(0);
	});

	it('changes to next index when next button is clicked', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];

		const componenet = shallow(
			<ImageCarousel images={images} />
		);

		//find next button and simulate a user click event.
		const nextButton = componenet.find('button').last();
		nextButton.simulate('click');

		expect(componenet.state().currentIndex).toEqual(1);
	});

	it('changes to previous index when prev button is clicked', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];

		const componenet = shallow(
			<ImageCarousel images={images} />
		);

		//find prev button and simulate a user click event.
		const prevButton = componenet.find('button').first();
		prevButton.simulate('click');

		expect(componenet.state().currentIndex).toEqual(5);
	});

	it('renders first image in array as first image displayed', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];
		const componenet = shallow(
			<ImageCarousel images={images} />
		);


		expect(componenet.find('img').props().src).toEqual('https://unsplash.it/500/?random');
	});

	it('changes to next image when next button is clicked', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];

		const componenet = shallow(
			<ImageCarousel images={images} />
		);

		//find prev button and simulate a user click event.
		const nextButton = componenet.find('button').last();
		nextButton.simulate('click');

		expect(componenet.find('img').props().src).toEqual('https://unsplash.it/501/?random');
	});

	it('changes to previous image when prev button is clicked', () => {

		const images = [
  			'https://unsplash.it/500/?random',
  			'https://unsplash.it/501/?random',
  			'https://unsplash.it/502/?random',
  			'https://unsplash.it/503/?random',
  			'https://unsplash.it/504/?random',
  			'https://unsplash.it/505/?random'
		];
		
		const componenet = shallow(
			<ImageCarousel images={images} />
		);

		//find prev button and simulate a user click event.
		const prevButton = componenet.find('button').first();
		prevButton.simulate('click');

		expect(componenet.find('img').props().src).toEqual('https://unsplash.it/505/?random');
	});

});