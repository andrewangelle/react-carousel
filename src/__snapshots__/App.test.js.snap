exports[`ImageCarousel renders without crashing 1`] = `
<div
  className="container"
  style={
    Object {
      "position": "relative",
      "width": undefined,
    }
  }>
  <div>
    Loading...
  </div>
</div>
`;
